### ISSUE TITLE
Unmatched {- does not allow - to be part of em/en-dash

### ISSUE THREAD

**Issue by hellux** [jgm/djot#125]
For the current implementation, if the opener is unmatched, the hyphen does not get parsed together with
later characters, e.g.

- `{---` generates `<p>{-&ndash;</p>` instead of `<p>{&mdash;</p>`,

- `{--` generates `<p>{--` instead of `<p>{&ndash;</p>`.

**Comment by jgm** [comment]
That's true; it's an artifact of a parsing strategy that tries not to backtrack.  It's not too worrisome because this is unlikely to occur naturally.

**Comment by dereuromark** [comment]
Should we close this then as not to be fixed?

**Comment by jgm** [comment]
I'm inclined to think yes.  @hellux how did this come up in real life?

**Comment by hellux** [comment]
I don't recall exactly when it came up, but I'm guessing there was just a difference in behavior between the reference implementation and the jotdown parser. However, they seem to be aligned right now.

**Comment by Omikhleia** [comment]
I dunno. It's **not** fixed until the specification includes it or rejects it, no?
And/or a few reference implementation exist (3 is usual in some standard committees ;) - which I understand https://github.com/jgm/djot.js/pull/121 is part of.

**Comment by dereuromark** [comment]
I agree. We should then reject with specs to clarify ideally.

**Comment by jgm** [comment]
Well, the current "spec" is more like a rough informal syntax description. We don't yet have the kind of spec I tried to produce for commonmark, with the aim of pinning down the meaning of any possible string and giving a pretty comprehensive set of tests.  I'm not saying we shouldn't aim to have such a thing, but the current syntax description isn't trying to be one.

**Comment by jgm** [comment]
djot.js is currently the reference implementation, which is not to say that it doesn't have bugs and unintended behavior.
